platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/lifecycle/StateMetadata.java

```diff
@@ -190,16 +187,36 @@ public static String validateIdentifier(@NonNull final String stateKey) {
     }
 
     /**
-     * Computes the label for a merkle node given the service name and state key
+     * Computes the label for a Merkle node given the service name and state key.
+     * <p>
+     * The label is computed as "serviceName.stateKey". The result is cached so that repeated calls
+     * with the same parameters return the same string without redoing the concatenation.
+     * </p>
      *
-     * @param serviceName The service name
-     * @param stateKey The state key
-     * @return The computed label
+     * @param serviceName the service name
+     * @param stateKey    the state key
+     * @return the computed label
      */
+    @Deprecated
     public static String computeLabel(@NonNull final String serviceName, @NonNull final String stateKey) {
         return Objects.requireNonNull(serviceName) + "." + Objects.requireNonNull(stateKey);
     }
 
+    /**
+     * Computes the label for a Merkle node given the service name and state ID.
+     * <p>
+     * The label is computed as "serviceName.stateId". The result is cached so that repeated calls
+     * with the same parameters return the same string without redoing the concatenation.
+     * </p>
+     *
+     * @param serviceName the service name
+     * @param stateId     the state ID
+     * @return the computed label
+     */
+    public static String computeLabel(@NonNull final String serviceName, final int stateId) {
```

Comment:
I have doubts about this method -- why not use just the state id?
(just pointing this out for offline discussion)